src/work_pool.c

```diff
@@ -206,25 +207,23 @@ work_pool_thread(void *arg)
 		 */
 		rc = pthread_cond_timedwait(&wpt->pqcond, &pool->pqh.qmutex,
 					    &ts);
-		if (rc) {
-			if (!wpt->work) {
-				/* Allow for possible timing race:
-				 * work entry can be submitted by another
-				 * thread during the timeout result?
-				 * Then, has already been removed there.
-				 * Only remove with no work here.
-				 */
-				pool->pqh.qcount--;
-				TAILQ_REMOVE(&pool->pqh.qh, &wpt->pqe, q);
-			}
-			if (unlikely(rc != ETIMEDOUT)) {
-				__warnx(TIRPC_DEBUG_FLAG_ERROR,
-					"%s() cond_timedwait failed (%d)\n",
-					__func__, rc);
-				break;
-			}
+		if (!wpt->work) {
+			/* Allow for possible timing race:
+			 * work entry can be submitted by another
+			 * thread during the thread task switch
+			 * after shutdown or timeout?
+			 * Then, has already been removed there.
+			 */
+			pool->pqh.qcount--;
+			TAILQ_REMOVE(&pool->pqh.qh, &wpt->pqe, q);
+		}
+		if (rc && rc != ETIMEDOUT) {
+			__warnx(TIRPC_DEBUG_FLAG_ERROR,
+				"%s() cond_timedwait failed (%d)\n",
+				__func__, rc);
+			break;
 		}
-	} while (wpt->work || pool->pqh.qcount <= pool->params.thrd_min);
+	} while (wpt->work || pool->pqh.qcount < pool->params.thrd_min);
```

Comment:
I think that `<=` was correct - if `pool->params.thrd_min` is 1 and there is only one thread, qcount will be 1 and thread will stop when there is no work? Or is qcount 0 when there's only the 0th thread?